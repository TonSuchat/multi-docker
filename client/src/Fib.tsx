import React, { useEffect, useState } from "react";
import axios from "axios";
import SeenIndexes from "./components/SeenIndexes";
import CalculatedValues from "./components/CalculatedValues";

export type FibState = {
  seenIndexes: any[];
  values: any;
  index: string;
};

const Fib = () => {
  const [state, setState] = useState<FibState>({
    seenIndexes: [],
    values: {},
    index: "",
  });

  useEffect(() => {
    const fetchValues = async () => {
      const values = await axios.get("/api/values/current");
      console.log("values", values);
      setState((s) => ({ ...s, values: values.data }));
    };
    const fetchIndexes = async () => {
      const seenIndexes = await axios.get("/api/values/all");
      setState((s) => ({ ...s, seenIndexes: seenIndexes.data }));
    };

    fetchValues();
    fetchIndexes();
  }, []);

  const handleSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
    await axios.post("/api/values", {
      index: state.index,
    });
    setState({ ...state, index: "" });
  };

  return (
    <div>
      <form onSubmit={handleSubmit}>
        <label>Enter your index:</label>
        <input
          type="number"
          value={state.index}
          onChange={(e) => setState({ ...state, index: e.target.value })}
        />
        <button>Submit</button>
      </form>
      <SeenIndexes data={state.seenIndexes} />
      <CalculatedValues values={state.values} />
    </div>
  );
};

export default Fib;
