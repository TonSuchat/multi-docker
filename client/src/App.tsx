import React from "react";
import { BrowserRouter as Router, Route, Link } from "react-router-dom";
import "./App.css";
import Fib from "./Fib";
import OtherPage from "./OtherPage";

const App: React.FC = () => {
  return (
    <Router>
      <div className="App">
        <header className="App-header">
          <Link className="App-link" to="/">
            Home
          </Link>
          <Link className="App-link" to="/otherpage">
            Other Page
          </Link>
        </header>
        <div style={{ marginTop: "2em" }}>
          <Route exact path="/" component={Fib} />
          <Route path="/otherpage" component={OtherPage} />
        </div>
      </div>
    </Router>
  );
};

export default App;
