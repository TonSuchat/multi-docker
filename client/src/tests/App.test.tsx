import { render } from "@testing-library/react";
import App from "../App";
import Fib from "../Fib";

jest.mock("../Fib");

describe("App", () => {
  let FibMock: jest.Mock;

  beforeEach(() => {
    FibMock = Fib as jest.Mock;
    FibMock.mockImplementation(() => <div>Fib</div>);
  });

  it("should match snapshot", () => {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("should generate links properly", () => {
    const { getByText } = render(<App />);
    expect(getByText("Home")).toBeInTheDocument();
    expect(getByText("Other Page")).toBeInTheDocument();
  });
});
