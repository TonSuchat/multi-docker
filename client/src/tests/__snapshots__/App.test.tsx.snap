// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot 1`] = `
<DocumentFragment>
  <header
    class="App-Header"
  >
    <a
      href="/"
    >
      Home
    </a>
    <a
      href="/otherpage"
    >
      Other Page
    </a>
  </header>
  <div>
    <div>
      Fib
    </div>
  </div>
</DocumentFragment>
`;
